invenio_records_resources/records/systemfields/pid.py

```diff
@@ -180,3 +182,69 @@ def dump_obj(field, record, pid):
 
         # Set ID on desired dictionary key.
         field.set_dictkey(record, pid.pid_value)
+
+
+class ModelPIDFieldContext(PIDFieldContext):
+    """Context for ModelPIDField.
+
+    This class implements the class-level methods available on a PIDField. I.e.
+    when you access the field through the class, for instance:
+
+    .. code-block:: python
+
+        Record.pid.resolve('...')
+        Record.pid.session_merge(record)
+    """
+
+    def resolve(self, pid_value, registered_only=True):
+        """Resolve identifier."""
+        resolver = self.field._resolver_cls(
+            self._record_cls, self.field.model_field_name
+        )
+        pid, record = resolver.resolve(pid_value)
+        self.field._set_cache(record, pid)
+
+        return record
+
+    def session_merge(self, record):
+        """Inactivate session merge since it all belongs to the same db obj."""
+        pass
+
+
+class ModelPIDField(ModelField):
+    """PID field in a db column on the record model. """
+
+    def __init__(
+        self,
+        model_field_name="pid",
+        resolver_cls=ModelResolver,
+        context_cls=ModelPIDFieldContext,
+    ):
+        """Initialise the dict field.
+        :param key: Name of key to store the pid value in.
+        """
+        self._resolver_cls = resolver_cls
+        self._context_cls = context_cls
+        super().__init__(model_field_name=model_field_name)
+
+    #
+    # Data descriptor
+    #
+    def __get__(self, record, owner=None):
+        """Accessing the attribute."""
+        # Class access
+        if record is None:
+            return self._context_cls(self, owner)
+        # Instance access
+        try:
+            return getattr(record.model, self.model_field_name)
+        except AttributeError:
+            return None
+
+    def pre_create(self, record):
```

Comment:
I do not understand why we need this. The `funder` record has a pid attr in the root, e.g.
```json
{
    "pid": ...
    "name": ...
}
```

Then when the record gets created, in invenio-records [here](https://github.com/inveniosoftware/invenio-records/blob/master/invenio_records/api.py#L73) the dict should get the attribute (since there is a `pid` field defined in the model). Which is how the other model fields work...

However, without this `pre_create` the tests fail because of a `pid` constraint null (i.e. the pid was not send to the DB in the INSERT query).